TheOracle2/IronswornRoller/IronswornRoll.cs

```diff
@@ -1,102 +1,115 @@
+            {
+                return IronswornRollOutcome.StrongHit;
+            }
+
+            if (Score <= ChallengeDie1.Value && Score <= ChallengeDie2.Value)
+            {
+                return IronswornRollOutcome.Miss;
+            }
+
+            return IronswornRollOutcome.WeakHit;
+        }
     }
-  }
-
-  public Color OutcomeColor() {
-    return Outcome switch {
-      IronswornRollOutcome.Miss => new Color(0xC50933),
-      IronswornRollOutcome.WeakHit => new Color(0x842A8C),
-      IronswornRollOutcome.StrongHit => new Color(0x47AEDD),
-      _ => new Color(0x842A8C),
+
+    public string OutcomeText()
+    {
+        if (IsMatch)
+        {
+            return Outcome switch
+            {
+                IronswornRollOutcome.Miss => IronswornRollResources.MissWithAMatch,
+                IronswornRollOutcome.WeakHit => IronswornRollResources.WeakHitWithAMatch,
+                IronswornRollOutcome.StrongHit => IronswornRollResources.StrongHitWithAMatch,
+                _ => "ERROR",
+            };
+        }
+        
+        return Outcome switch
+        {
+            IronswornRollOutcome.Miss => IronswornRollResources.Miss,
+            IronswornRollOutcome.WeakHit => IronswornRollResources.Weak_Hit,
+            IronswornRollOutcome.StrongHit => IronswornRollResources.Strong_Hit,
+            _ => "ERROR",
+        };
+    }
+
+    public Color OutcomeColor() => Outcome switch
+    {
+        IronswornRollOutcome.Miss => new Color(0xC50933),
+        IronswornRollOutcome.WeakHit => new Color(0x842A8C),
+        IronswornRollOutcome.StrongHit => new Color(0x47AEDD),
+        _ => new Color(0x842A8C),
     };
-  }
-
-  public string OutcomeIcon() {
-    return Outcome switch {
-      IronswornRollOutcome.Miss => IronswornRollResources.MissImageURL,
-      IronswornRollOutcome.WeakHit => IronswornRollResources.WeakHitImageURL,
-      IronswornRollOutcome.StrongHit => IronswornRollResources.StrongHitImageURL,
-      _ => IronswornRollResources.MissImageURL,
+
+    public string OutcomeIcon() => Outcome switch
+    {
+        IronswornRollOutcome.Miss => IronswornRollResources.MissImageURL,
+        IronswornRollOutcome.WeakHit => IronswornRollResources.WeakHitImageURL,
+        IronswornRollOutcome.StrongHit => IronswornRollResources.StrongHitImageURL,
+        _ => IronswornRollResources.MissImageURL,
     };
-  }
-
-  public EmbedBuilder ToEmbed() {
-    return new EmbedBuilder()
-        .WithColor(OutcomeColor())
-        .WithThumbnailUrl(OutcomeIcon())
-        .WithTitle(OutcomeText())
-        .WithDescription(ToString())
-        .WithFooter(FooterText())
-        .WithAuthor(AuthorText);
-  }
-
-  public string OverMaxMessage() {
-    return RawScore > 10 ? string.Format(IronswornRollResources.OverMaxMessage, Score) : "";
-  }
-  public virtual string FooterText() {
-    return OverMaxMessage();
-  }
-  public virtual string AuthorText {
```

Comment:
yeah, `OverMaxMessage()` already points to the localizable string:
<img width="926" alt="Screen Shot 2022-01-03 at 3 03 35 PM" src="https://user-images.githubusercontent.com/5354757/147989731-ecc9c78a-89a9-4a0e-8a54-6ebfd82623af.png">

rationale: i'm trying to keep "what goes in the message embed footer?" and "what text should be shown if the RawScore exceeds 10?" separate, since they're technically different questions even if their answer is *currently* the same -- it might change in the future if we decide to display that text elsewhere in the footer, or to display roll results in something other than a message embed.

though i guess to be consistent with that line of thinking, i ought to follow a similar pattern with AuthorText